constellations/orion/src/repos/account_repo.go

```diff
@@ -117,32 +118,82 @@ func (acc *accountRepo) SelectAllNegativeBalances() ([]domains.AccountSum, error
 	return results, nil
 }
 
-func (acc *accountRepo) Insert(account domains.Account) error {
+func (acc *accountRepo) InsertWithUser(account domains.Account, user domains.User) error {
+	tx, err := acc.db.Begin()
+	if err != nil {
+		return err
+	}
 	utils.LogWithContext("accountRepo.Insert", logger.Fields{"account": account})
 	statement := "INSERT INTO accounts (" +
 		"created_at, " +
 		"updated_at, " +
 		"primary_email, " +
 		"password" +
 		") VALUES (?, ?, ?, ?)"
-
 	stmt, err := acc.db.Prepare(statement)
 	if err != nil {
+		tx.Rollback()
 		return appErrors.WrapDbPrepare(err, statement)
 	}
 	defer stmt.Close()
 
 	now := time.Now().UTC()
-	execResult, err := stmt.Exec(
+	_, err = stmt.Exec(
 		now,
 		now,
 		account.PrimaryEmail,
 		account.Password,
 	)
 	if err != nil {
-		return appErrors.WrapDbExec(err, statement, account)
+		tx.Rollback()
+		return err
```

Comment:
We don't want to just return `err` anymore. We want to wrap it with some context which is what the below function does.
`appErrors.WrapDbExec(err, statement, account)`

We're going to wrap it with a `ERR_DB_EXEC` so we know it's an error that relates to DB Executing and we also pass in information of what we were trying to execute. Check out the appErrors package!